build/source/dshare/globalData.f90

```diff
@@ -0,0 +1,118 @@
+ USE var_lookup,only:maxvarDiag      ! diagnostic variables:     maximum number variables
+ USE var_lookup,only:maxvarFlux      ! model fluxes:             maximum number variables
+ USE var_lookup,only:maxvarDeriv     ! model derivatives:        maximum number variables
+ USE var_lookup,only:maxvarIndx      ! model indices:            maximum number variables
+ USE var_lookup,only:maxvarMpar      ! model parameters:         maximum number variables
+ USE var_lookup,only:maxvarBvar      ! basin-average variables:  maximum number variables
+ USE var_lookup,only:maxvarBpar      ! basin-average parameters: maximum number variables
+ USE var_lookup,only:maxvarDecisions ! maximum number of decisions
+ implicit none
+ private
+
+ ! Define the model decisions
+ type(model_options),save,public             :: model_decisions(maxvarDecisions)  ! the model decision structure
+
+ ! Define metadata for model forcing datafile
+ type(file_info),save,public,allocatable     :: forcFileInfo(:)         ! file info for model forcing data
+
+ ! define default parameter values and parameter bounds 
+ type(par_info),save,public                  :: localParFallback(maxvarMpar) ! local column default parameters
+ type(par_info),save,public                  :: basinParFallback(maxvarBpar) ! basin-average default parameters
+
+ ! define vectors of metadata
+ type(var_info),save,public                  :: time_meta(maxvarTime)   ! model time information
+ type(var_info),save,public                  :: forc_meta(maxvarForc)   ! model forcing data
+ type(var_info),save,public                  :: attr_meta(maxvarAttr)   ! local attributes
+ type(var_info),save,public                  :: type_meta(maxvarType)   ! local classification of veg, soil, etc.
+ type(var_info),save,public                  :: mpar_meta(maxvarMpar)   ! local model parameters for each HRU
+ type(var_info),save,public                  :: indx_meta(maxvarIndx)   ! local model indices for each HRU
+ type(var_info),save,public                  :: prog_meta(maxvarProg)   ! local state variables for each HRU
+ type(var_info),save,public                  :: diag_meta(maxvarDiag)   ! local diagnostic variables for each HRU
+ type(var_info),save,public                  :: flux_meta(maxvarFlux)   ! local model fluxes for each HRU
+ type(var_info),save,public                  :: deriv_meta(maxvarDeriv) ! local model derivatives for each HRU
+ type(var_info),save,public                  :: bpar_meta(maxvarBpar)   ! basin parameters for aggregated processes
+ type(var_info),save,public                  :: bvar_meta(maxvarBvar)   ! basin variables for aggregated processes
+
+ ! ancillary metadata structures
+ type(extended_info),save,public,allocatable :: averageFlux_meta(:)     ! timestep-average model fluxes
+
+ ! define summary information on all data structures
+ integer(i4b),parameter                      :: nStruct=12              ! number of data structures
```

Comment:
Is there a way to calculate `nStruct` programmatically (so that it automatically updates when we add another element to `structInfo`?